src/util/battery/batterywindows.cpp

```diff
@@ -44,6 +40,10 @@ void BatteryWindows::read() {
         if (seconds_left >= 0) {
             m_iMinutesLeft = seconds_left / 60;
```

Comment:
I don't remember that I ever suggested such a change ;)

`m_iMinutesLeft` is not always set in this branch. Initializing both `m_chargingState` and `m_iMinutesLeft` with *UNKNOWN* helps to guarantee a consistent state upon exit.

Reply:
Setting m_dPercentage to 0.0 should not even be required if m_chargingState = Battery::UNKNOWN? Initializing it with NaN instead of 0.0 would emphasize this.

Considering the complexity of the various branches I would recommend to keep the initialization to invalid/unknown values at the beginning of the function.

Reply:
> I don't remember that I ever suggested such a change ;)

Sorry, it wasn't you, @daschuer requested this change in https://github.com/mixxxdj/mixxx/pull/2160:

"Can you move that into the else branch below. This way the member value is always valid."

Feel free to close this pull request if you think the change is unnecessary.

Reply:
> `m_iMinutesLeft` is not always set in this branch. Initializing both `m_chargingState` and `m_iMinutesLeft` with _UNKNOWN_ helps to guarantee a consistent state upon exit.

I have fixed the not always set `m_iMinutesLeft` and `m_chargingState`. 
Not initializing them with UNKNOWN is the point of this pull request. 
(Setting `m_chargingState` to UNKNOWN for a short period of time, if it is actually known may be suboptimal)

Reply:
No interception possible, the unknown value will never become visible. Control flows could be simplified if only known and defined values need to be written. And it is safer if you could exit the function at any time while still leaving the object in a valid state.

Reply:
> No interception possible

Yes, there is no interception possible. The unknown value could only become visible in a crash dump or debug session. Not sure if it really matters..